test/unit/specs/core/createEventManager.spec.js

```diff
@@ -108,31 +108,51 @@ describe("createEventManager", () => {
         });
     });
 
-    it("sets the lastChanceCallback, which wraps config.onBeforeEventSend, on the event", () => {
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("events call onBeforeEventSend callback", () => {
+      event.finalize.and.callFake(onBeforeEventSend => {
+        onBeforeEventSend();
       });
-      return eventManager.sendEvent(event, {}).then(() => {
-        wrappedLastChanceCallback();
+      return eventManager.sendEvent(event).then(() => {
         expect(config.onBeforeEventSend).toHaveBeenCalled();
       });
     });
 
-    it("logs errors in the config.onBeforeEventSend callback", () => {
-      const error = new Error("onBeforeEventSend error");
-      config.onBeforeEventSend.and.throwError(error);
+    it("returns false onBeforeEventSend and event should not be sent", () => {
```

Comment:
From a unit perspective, this isn't really testing what happens when `onBeforeEventSend` returns `false`. I think a more appropriate description would be: `does not send event when event.shouldSend returns false`. The test for whether `event.shouldSend` returns `false` when `onBeforeEventSend` returns `false` belongs in createEvent.spec.js.